### PR TITLE
[test] lr_find with bs_scale

### PR DESCRIPTION
**Pull request by s-rog**
## What does this PR do?

add test for #4688
(should have included this with 4688, apologies)

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/trainer/test_lr_finder.py [hunk 1]
```diff
@@ -271,3 +271,22 @@ def test_lr_finder_fails_fast_on_bad_config(tmpdir):
...
     with pytest.raises(MisconfigurationException, match='should have one of these fields'):
         trainer.tune(BoringModel())
+
+
+def test_lr_find_with_bs_scale(tmpdir):
+    """ Test that lr_find runs with batch_size_scaling """
+
+    hparams = EvalModelTemplate.get_default_hparams()
+    before_lr = hparams.get('learning_rate')
+    model = EvalModelTemplate(**hparams)
```
Comment (carmocca) [inline tests/trainer/test_lr_finder.py:281]: Can we use `BoringModel` instead? It's much faster

Otherwise `ClassificationModel`

**Review by tchaton — APPROVED** [review]
LGTM !

File: tests/trainer/test_lr_finder.py [hunk 2]
```diff
@@ -271,3 +271,27 @@ def test_lr_finder_fails_fast_on_bad_config(tmpdir):
...
+def test_lr_find_with_bs_scale(tmpdir):
+    """ Test that lr_find runs with batch_size_scaling """
+
+    class BoringModelTune(BoringModel):
+        def __init__(self, hparams):
+            super().__init__()
+            self.save_hyperparameters(hparams)
+
+    hparams = EvalModelTemplate.get_default_hparams()
+    before_lr = hparams.get('learning_rate')
```
Comment (carmocca) [inline tests/trainer/test_lr_finder.py:285]: Why don't you hardcode a `lr` in `BoringModelTune` instead of getting it from `EvalModelTemplate`?

Comment (s-rog) [reply tests/trainer/test_lr_finder.py:285]: the hparams contains bs for bs scaling as well

Comment (carmocca) [reply tests/trainer/test_lr_finder.py:285]: I mean something like


```python
    class BoringModelTune(BoringModel):
        def __init__(self, lr=0.01):
            super().__init__()
            self.lr = lr
            self.save_hyperparameters(hparams)

    before_lr = model.lr
```

Just to not rely on `EvalModelTemplate` at all

Comment (s-rog) [reply tests/trainer/test_lr_finder.py:285]: Yeah can do that as well, just thought I'd reuse existing code since the rest of that script uses EvalModelTemplate. I'll update the PR as soon as I get a chance.

**Review by awaelchli — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]
